File: GdPicture/WinForm/vb.net/PDF to TIFF/Form1.vb
Option Explicit On
Option Strict On

Imports GdPicture14

Public Class Form1

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        OpenFileDialog1.FileName = ""
        OpenFileDialog1.Filter = "PDF Document (*.pdf)|*.pdf"
        OpenFileDialog1.Multiselect = False
        If OpenFileDialog1.ShowDialog() = Windows.Forms.DialogResult.OK Then
            TextBox1.Text = OpenFileDialog1.FileName
        End If
    End Sub


    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Dim oLicenseManager As New GdPicture14.LicenseManager() 'Go to http://www.gdpicture.com/download-gdpicture/ to get a 1 month trial key unlocking all features of the toolkit.
        oLicenseManager.RegisterKEY("XXXX") 'Please, replace XXXX by a valid demo or commercial license key.
        ComboBox1.Text = "AUTO"
    End Sub


    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        If TextBox1.Text <> "" Then

            Dim MultiTiffID As Integer
            Dim CompressionScheme As GdPicture14.TiffCompression
            Dim Status As GdPicture14.GdPictureStatus = GdPicture14.GdPictureStatus.OK

            Select Case ComboBox1.Text
                Case "AUTO"
                    CompressionScheme = GdPicture14.TiffCompression.TiffCompressionAUTO
                Case "JPEG"
                    CompressionScheme = GdPicture14.TiffCompression.TiffCompressionJPEG
                Case "CCITT3"
                    CompressionScheme = GdPicture14.TiffCompression.TiffCompressionCCITT3
                Case "CCITT4"
                    CompressionScheme = GdPicture14.TiffCompression.TiffCompressionCCITT4
                Case "LZW"
                    CompressionScheme = GdPicture14.TiffCompression.TiffCompressionLZW
                Case "RLE"
                    CompressionScheme = GdPicture14.TiffCompression.TiffCompressionRLE
                Case "NONE"
                    CompressionScheme = GdPicture14.TiffCompression.TiffCompressionNONE
                Case "ZIP"
                    CompressionScheme = GdPicture14.TiffCompression.TiffCompressionDEFLATE
            End Select


            Using oGdPicturePDF As New GdPicture14.GdPicturePDF
                Dim InputFilePath As String = TextBox1.Text
                If oGdPicturePDF.LoadFromFile(InputFilePath, False) = GdPicture14.GdPictureStatus.OK Then
                    ProgressBar1.Maximum = oGdPicturePDF.GetPageCount
                    Dim DPI As Integer = CInt(Val(TextBox2.Text))
                    Dim OutputFilePath As String = Mid(InputFilePath, 1, Len(InputFilePath) - 3) + "tif"

                    Using oGdPictureImaging As New GdPicture14.GdPictureImaging
                        For i As Integer = 1 To oGdPicturePDF.GetPageCount
                            If Status = GdPicture14.GdPictureStatus.OK Then
                                oGdPicturePDF.SelectPage(i)
                                Dim RasterizedPageID As Integer = oGdPicturePDF.RenderPageToGdPictureImageEx(DPI, True)

                                'CCITT3 & CCITT4 compression support only bitonal (1 bpp) images !!
                                If CompressionScheme = GdPicture14.TiffCompression.TiffCompressionCCITT3 Or CompressionScheme = GdPicture14.TiffCompression.TiffCompressionCCITT4 Then
                                    oGdPictureImaging.ConvertTo1BppSauvola(RasterizedPageID, 0.35)
                                End If

                                If i = 1 Then
                                    MultiTiffID = RasterizedPageID 'Warning: this image must be released at the end!!!!
                                    Status = oGdPictureImaging.TiffSaveAsMultiPageFile(MultiTiffID, OutputFilePath, CompressionScheme)
                                Else
                                    Status = oGdPictureImaging.TiffAddToMultiPageFile(MultiTiffID, RasterizedPageID)
                                    oGdPictureImaging.ReleaseGdPictureImage(RasterizedPageID)
                                End If
                            End If
                            ProgressBar1.Value = i
                        Next

                        oGdPictureImaging.TiffCloseMultiPageFile(MultiTiffID)
                        oGdPictureImaging.ReleaseGdPictureImage(MultiTiffID)

                    End Using

                    If Status = GdPicture14.GdPictureStatus.OK Then
                        MsgBox("The image " + OutputFilePath + " has been correctly created")
                    Else
                        MsgBox("Error: " + Status.ToString)
                    End If
                    oGdPicturePDF.CloseDocument()
                Else
                    MsgBox("Can't open file: " & InputFilePath)
                End If
            End Using
        Else
            MsgBox("Please, select a file to convert.")
        End If
    End Sub
End Class

File: GdPicture/WinForm/C#/PDF to TIFF/Form1.cs

using System.Collections.Generic;
using System;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using System.Collections;
using System.Windows.Forms;


using GdPicture14;


namespace PDF_to_TIFF
{
	
	public partial class Form1
	{
		public Form1()
		{
			InitializeComponent();
			
			//Added to support default instance behaviour in C#
			if (defaultInstance == null)
				defaultInstance = this;
		}
		
#region Default Instance
		
		private static Form1 defaultInstance;
		
		/// <summary>
		/// Added by the VB.Net to C# Converter to support default instance behavour in C#
		/// </summary>
public static Form1 Default
		{
			get
			{
				if (defaultInstance == null)
				{
					defaultInstance = new Form1();
					defaultInstance.FormClosed += new FormClosedEventHandler(defaultInstance_FormClosed);
				}
				
				return defaultInstance;
			}
			set
			{
				defaultInstance = value;
			}
		}
		
		static void defaultInstance_FormClosed(object sender, FormClosedEventArgs e)
		{
			defaultInstance = null;
		}
		
#endregion
		
		public void Button2_Click(System.Object sender, System.EventArgs e)
		{
			OpenFileDialog1.FileName = "";
			OpenFileDialog1.Filter = "PDF Document (*.pdf)|*.pdf";
			OpenFileDialog1.Multiselect = false;
			if (OpenFileDialog1.ShowDialog() == System.Windows.Forms.DialogResult.OK)
			{
				TextBox1.Text = OpenFileDialog1.FileName;
			}
		}
		
		
		public void Form1_Load(System.Object sender, System.EventArgs e)
		{
			LicenseManager oLicenseManager = new LicenseManager(); //Go to http://www.gdpicture.com/download-gdpicture/ to get a 1 month trial key unlocking all features of the toolkit.
			oLicenseManager.RegisterKEY("XXX"); //Please, replace XXX by a valid demo or commercial license key.
			ComboBox1.Text = "AUTO";
		}
		
		
		public void Button1_Click(System.Object sender, System.EventArgs e)
		{
			if (TextBox1.Text != "")
			{
				int multiTiffID = 0;

                TiffCompression compressionScheme ;
				GdPictureStatus status = GdPictureStatus.OK;
				
				switch (ComboBox1.Text)
				{
					case "AUTO":
                    default:
						compressionScheme = TiffCompression.TiffCompressionAUTO;
						break;
					case "JPEG":
						compressionScheme = TiffCompression.TiffCompressionJPEG;
						break;
					case "CCITT3":
						compressionScheme = TiffCompression.TiffCompressionCCITT3;
						break;
					case "CCITT4":
						compressionScheme = TiffCompression.TiffCompressionCCITT4;
						break;
					case "LZW":
						compressionScheme = TiffCompression.TiffCompressionLZW;
						break;
					case "RLE":
						compressionScheme = TiffCompression.TiffCompressionRLE;
						break;
					case "NONE":
						compressionScheme = TiffCompression.TiffCompressionNONE;
						break;
					case "ZIP":
						compressionScheme = GdPicture14.TiffCompression.TiffCompressionDEFLATE;
						break;
				}
				
				
				using (GdPicturePDF oGdPicturePDF = new GdPicturePDF())
				{
					string InputFilePath = TextBox1.Text;
					if (oGdPicturePDF.LoadFromFile(InputFilePath, false) == GdPictureStatus.OK)
					{
						ProgressBar1.Maximum = oGdPicturePDF.GetPageCount();

                        int DPI = int.Parse(TextBox2.Text);
						string OutputFilePath = InputFilePath.Substring(0, InputFilePath.Length - 3) + "tif";
						
						using (GdPictureImaging oGdPictureImaging = new GdPictureImaging())
						{
							for (int i = 1; i <= oGdPicturePDF.GetPageCount(); i++)
							{
								if (status == GdPictureStatus.OK)
								{
									oGdPicturePDF.SelectPage(i);
									int RasterizedPageID = oGdPicturePDF.RenderPageToGdPictureImageEx(DPI, true);
									
									//CCITT3 & CCITT4 compression support only bitonal (1 bpp) images !!
									if (compressionScheme == TiffCompression.TiffCompressionCCITT3 || compressionScheme == TiffCompression.TiffCompressionCCITT4)
									{
										oGdPictureImaging.ConvertTo1BppAT(RasterizedPageID);
									}
									
									if (i == 1)
									{
										multiTiffID = RasterizedPageID; //Warning: this image must be released at the end!!!!
										status = oGdPictureImaging.TiffSaveAsMultiPageFile(multiTiffID, OutputFilePath, compressionScheme);
									}
									else
									{
										status = oGdPictureImaging.TiffAddToMultiPageFile(multiTiffID, RasterizedPageID);
										oGdPictureImaging.ReleaseGdPictureImage(RasterizedPageID);
									}
								}
								ProgressBar1.Value = i;
							}
							
							oGdPictureImaging.TiffCloseMultiPageFile(multiTiffID);
							oGdPictureImaging.ReleaseGdPictureImage(multiTiffID);
						}
						
						
						if (status == GdPictureStatus.OK)
						{
							MessageBox.Show("The image " + OutputFilePath + " has been correctly created");
						}
						else
						{
							MessageBox.Show("Error: " + status.ToString());
						}
						oGdPicturePDF.CloseDocument();
					}
					else
					{
						MessageBox.Show("Can\'t open file: " + InputFilePath);
					}
				}
				
			}
			else
			{
				MessageBox.Show("Please, select a file to convert.");
			}
		}
	}
}
